dist/web3-light.js

```diff
@@ -3495,6 +3516,201 @@ module.exports = ICAP;
     You should have received a copy of the GNU Lesser General Public License
     along with ethereum.js.  If not, see <http://www.gnu.org/licenses/>.
 */
+/** @file ipcprovider.js
+ * @authors:
+ *   Fabian Vogelsteller <fabian@ethdev.com>
+ * @date 2015
+ */
+
+"use strict";
+
+var utils = require('../utils/utils');
+var errors = require('./errors');
+
+var errorTimeout = '{"jsonrpc": "2.0", "error": {"code": -32603, "message": "IPC Request timed out for method  \'__method__\'"}, "id": "__id__"}';
+
+
+var IpcProvider = function (path, net) {
+    var _this = this;
+    this.responseCallbacks = {};
+    this.path = path;
+    
+    net = net || require('net');
+
+    this.connection = net.connect({path: this.path});
+
+    this.connection.on('error', function(e){
+        console.error('IPC Connection Error', e);
+        _this._timeout();
+    });
+
+    this.connection.on('end', function(){
+        _this._timeout();
+    }); 
+
+
+    // LISTEN FOR CONNECTION RESPONSES
+    this.connection.on('data', function(data) {
+        /*jshint maxcomplexity: 6 */
+        data = data.toString();
+
+        // DE-CHUNKER
+        var dechunkedData = data
+            .replace(/\}\{/g,'}|--|{') // }{
+            .replace(/\}\]\[\{/g,'}]|--|[{') // }][{
+            .replace(/\}\[\{/g,'}|--|[{') // }[{
+            .replace(/\}\]\{/g,'}]|--|{') // }]{
+            .split('|--|');
+
+
+        dechunkedData.forEach(function(data){
+
+            // prepend the last chunk
+            if(_this.lastChunk)
+                data = _this.lastChunk + data;
```

Comment:
because a request can be half complete if the last buffer was full, so i se the `_this.lastChunk` and prepend it to the next buffer piece i get on the next call of `on('data'`.

This is all tested and works.